sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -0,0 +1,358 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// https://azure.github.io/azure-sdk/typescript_design.html#ts-config-lib
+/// <reference lib="esnext.asynciterable" />
+
+import { AppConfigCredential } from "./appConfigCredential";
+import { TokenCredential } from "@azure/core-http";
+import { AppConfiguration } from "./generated/src/appConfiguration";
+import { PageSettings, PagedAsyncIterableIterator } from "@azure/core-paging";
+import "@azure/core-asynciterator-polyfill";
+
+import {
+  AddConfigurationSettingResponse,
+  ClearReadOnlyOptions,
+  ConfigurationSetting,
+  ConfigurationSettingOptions,
+  ConfigurationSettingParam,
+  DeleteConfigurationSettingOptions,
+  DeleteConfigurationSettingResponse,
+  ETagOption,
+  GetConfigurationSettingOptions,
+  GetConfigurationSettingResponse,
+  GetKeyValuesResponse,
+  ListConfigurationSettingPage,
+  ListConfigurationSettingsOptions,
+  ListRevisionsOptions,
+  ListRevisionsPage,
+  SetConfigurationSettingResponse,
+  SetReadOnlyOptions,
+  SetReadOnlyResponse,
+  ClearReadOnlyResponse,
+} from "./models";
+import { formatETagForMatchHeaders, formatWildcards, extractAfterTokenFromNextLink } from './internal/helpers';
+
+const apiVersion = "1.0";
+const ConnectionStringRegex = /Endpoint=(.*);Id=(.*);Secret=(.*)/;
+const deserializationContentTypes = {
+  json: [
+    "application/vnd.microsoft.appconfig.kvset+json",
+    "application/vnd.microsoft.appconfig.kv+json",
+    "application/vnd.microsoft.appconfig.kvs+json",
+    "application/vnd.microsoft.appconfig.keyset+json",
+    "application/vnd.microsoft.appconfig.revs+json"
+  ]
+};
+
+/**
+ * Client for the Azure App Configuration service.
+ */
+export class AppConfigurationClient {
+  private client: AppConfiguration;
+
+  /**
+   * Initializes a new instance of the AppConfigurationClient class.
+   * @param connectionString Connection string needed for a client to connect to Azure.
+   */
+  constructor(connectionString: string);
+  /**
+   * Initializes a new instance of the AppConfigurationClient class.
+   * @param uri The base URI for the Azure service
+   * @param credential The credentials to use for authentication.
+   */
+  constructor(uri: string, credential: TokenCredential);
+  constructor(uriOrConnectionString: string, credential?: TokenCredential) {
+    if (uriOrConnectionString == null) {
+      throw new Error(
+        "You must provide a connection string or the URL for your AppConfiguration instance"
+      );
+    }
+
+    const regexMatch = uriOrConnectionString.match(ConnectionStringRegex);
+    if (regexMatch) {
+      const credential = new AppConfigCredential(regexMatch[2], regexMatch[3]);
+      this.client = new AppConfiguration(credential, apiVersion, {
+        baseUri: regexMatch[1],
+        deserializationContentTypes
+      });
+    } else if (credential && credential.constructor.name === "ManagedIdentityCredential") {
```

Comment:
if it only accepts `ManagedIdentityCredentials` why make it take `TokenCredential`? Seems like it should take the more specific type as an argument.

Reply:
Hm...that's a good point. @chradek  - I think this was in here from the beginning - any reason why we can't make the type a bit more specific here? Was it future expansion?

Reply:
@daviwil 
Asking since you wrote this part originally. Though future-expansion sounds reasonable...

Reply:
Maybe it's better to just change it to `ManagedIdentityCredential` for now and then change it back to `TokenCredential` later if it turns out that the AppConfig service expands to take more AAD credential types.

Reply:
After talking with everyone I realized that this code path isn't going to work either way. I'm removing it for now and we'll come back to it when it's ready at the service level.